Bottom: 2b4d6d06f76ea2552d4edc06628120101d23bb70
Top:    363b4fc01b9f8b56bc5e05642ba7f8f62e8a4298
Author: Ian Kent <raven@themaw.net>
Date:   2013-08-02 16:48:05 +0800

Refresh of autofs.debug

---

diff --git a/modules/lookup_file.c b/modules/lookup_file.c
index 087b4dd..e9697e8 100644
--- a/modules/lookup_file.c
+++ b/modules/lookup_file.c
@@ -1141,11 +1141,13 @@ do_cache_lookup:
 	debug(ap->logopt, MODPREFIX "%s -> %s", key, mapent);
 	ret = ctxt->parse->parse_mount(ap, key, key_len,
 				       mapent, ctxt->parse->context);
-	/* Don't update negative cache when attempting to re-connect */
-	if (ret && !(ap->flags & MOUNT_FLAG_REMOUNT)) {
-		cache_writelock(mc);
-		cache_update_negative(mc, source, key, ap->negative_timeout);
-		cache_unlock(mc);
+	if (ret) {
+		/* Don't update negative cache when re-connecting */
+		if (!(ap->flags & MOUNT_FLAG_REMOUNT)) {
+			cache_writelock(mc);
+			cache_update_negative(mc, source, key, ap->negative_timeout);
+			cache_unlock(mc);
+		}
 		return NSS_STATUS_TRYAGAIN;
 	}
 
diff --git a/modules/lookup_ldap.c b/modules/lookup_ldap.c
index a388433..fb8de9a 100644
--- a/modules/lookup_ldap.c
+++ b/modules/lookup_ldap.c
@@ -3062,11 +3062,13 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 	debug(ap->logopt, MODPREFIX "%s -> %s", key, mapent);
 	ret = ctxt->parse->parse_mount(ap, key, key_len,
 				       mapent, ctxt->parse->context);
-	/* Don't update negative cache when attempting to re-connect */
-	if (ret && !(ap->flags & MOUNT_FLAG_REMOUNT)) {
-		cache_writelock(mc);
-		cache_update_negative(mc, source, key, ap->negative_timeout);
-		cache_unlock(mc);
+	if (ret) {
+		/* Don't update negative cache when re-connecting */
+		if (!(ap->flags & MOUNT_FLAG_REMOUNT)) {
+			cache_writelock(mc);
+			cache_update_negative(mc, source, key, ap->negative_timeout);
+			cache_unlock(mc);
+		}
 		return NSS_STATUS_TRYAGAIN;
 	}
